smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/ContextHolder.java

```diff
@@ -0,0 +1,39 @@
+package io.smallrye.reactive.messaging.kafka.commit;
+
+import io.vertx.mutiny.core.Context;
+import io.vertx.mutiny.core.Vertx;
+
+/**
+ * A class holding a vert.x context to make sure methods are always run from the same one.
+ */
+public class ContextHolder {
+
+    protected final Vertx vertx;
+    protected volatile Context context;
+
+    public ContextHolder(Vertx vertx) {
+        this.vertx = vertx;
+    }
+
+    public Context getContext() {
+        Context ctx = this.context;
+        if (ctx == null) {
+            synchronized (this) {
+                ctx = this.context;
+                if (ctx == null) {
+                    this.context = ctx = vertx.getOrCreateContext();
+                }
+            }
+        }
+        return ctx;
+    }
```

Comment:
This pattern comes from
https://www.oracle.com/technical-resources/articles/javase/bloch-effective-08-qa.html

Under: Best Practices for Lazy Initialization

It's also referenced in Java Concurrency in Practice by Goetz et al in section 16.2.4

> Subsequent changes in the JMM... have enabled DLC to work if resource is made volatile.

Reply:
Oh, I know the pattern, it was just late at night :-D 
I isolated it into its own class, so we can reuse the pattern.

Going to be fun with the rebalance listener not called on the event loop anymore...

Reply:
I have no doubt you knew :D

But I assume others might read this.

Reply:
> Going to be fun with the rebalance listener not called on the event loop anymore...

Yes! Haven't looked at your draft PR yet.

Reply:
I also admit to having used the DLC before Java 5. I was younger and naive 🤣